mdata/notifierKafka/notifierKafka.go

```diff
@@ -93,11 +96,33 @@ func (c *NotifierKafka) start() {
 		if err != nil {
 			log.Fatal(4, "kafka-cluster: Failed to get %q duration offset for %s:%d. %q", offsetStr, topic, partition, err)
 		}
-		go c.consumePartition(topic, partition, offset)
+		processBacklog.Add(1)
+		go c.consumePartition(topic, partition, offset, processBacklog)
 	}
+	// wait for our backlog to be processed before returning.  This will block metrictank from consuming metrics until
+	// we have processed old metricPersist messages. The end result is that we wont overwrite chunks in cassandra that
+	// have already been previously written.
+	// We dont wait more then 60seconds for the backlog to be processed.
+	log.Info("kafka-cluster: waiting for metricPersist backlog to be processed.")
+	backlogProcessed := make(chan struct{}, 1)
+	go func() {
+		processBacklog.Wait()
+		backlogProcessed <- struct{}{}
+	}()
+	ticker := time.NewTicker(backlogProcessTimeout)
+
+	select {
+	case <-ticker.C:
```

Comment:
no need for a ticker. just do channel read from `time.After(backlogProcessTimeout)`